src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -241,9 +241,9 @@ class MacroAssembler: public Assembler {
 
   // idiv variant which deals with MINLONG as dividend and -1 as divisor
   int corrected_idivl(Register result, Register rs1, Register rs2,
-                      bool want_remainder);
+                      bool want_remainder, bool is_signed = true);
```

Comment:
Could you not set the default value of `is_signed` to `true`, to make it clear which case it is at the callsite.